indexers/staking-squid/src/utils/cache.ts

```diff
@@ -0,0 +1,63 @@
+import type { Store } from "@subsquid/typeorm-store";
+import {
+  Account,
+  Deposit,
+  Domain,
+  Nominator,
+  Operator,
+  Stats,
+  StatsPerDomain,
+  StatsPerOperator,
+  Withdrawal,
+} from "../model";
+import type { Ctx } from "../processor";
+
+export type Cache = {
+  domains: Map<string, Domain>;
+  accounts: Map<string, Account>;
+  operators: Map<string, Operator>;
+  nominators: Map<string, Nominator>;
+  deposits: Map<string, Deposit>;
+  withdrawals: Map<string, Withdrawal>;
+
+  stats: Map<string, Stats>;
+  statsPerDomain: Map<string, StatsPerDomain>;
+  statsPerOperator: Map<string, StatsPerOperator>;
+};
+
+export const initCache: Cache = {
+  domains: new Map(),
+  accounts: new Map(),
+  operators: new Map(),
+  nominators: new Map(),
+  deposits: new Map(),
+  withdrawals: new Map(),
+
+  stats: new Map(),
+  statsPerDomain: new Map(),
+  statsPerOperator: new Map(),
+};
+
+export const load = async (ctx: Ctx<Store>): Promise<Cache> => {
```

Comment:
If I understand correctly, the entire squid will be in the cache. Do we have any concerns about this or am I not following correctly?

Reply:
Not the entire squid, just 4 tables are  needed (Domain, Account, Operator and Nominator

My assumption is that loading everything ounce is better than to check if the entry exist each time...